linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -331,24 +334,47 @@
     loop_var: definition
   ignore_errors: no
 
-- name: "mac_and_ip | extract mac address"
-  shell: >
-    virsh -c {{ node[1] }} dumpxml {{ node[0] }}_{{ node[2] }}
-    | grep 'mac address'
-    | cut -f 2 -d "'"
+- name: "dump node data"
+  virt:
+    name: "{{ data[0] }}_{{ data[2] }}"
+    command: get_xml
+    uri: "{{ data[1] }}"
   with_nested:
     - ["{{ libvirt_resource_name }}"]
     - ["{{ res_def['uri'] | default('qemu:///system') }}"]
     - "{{ res_count }}"
+  loop_control:
+    loop_var: data
+  when: not async
+  register: node_data
+
+- xml:
+    xmlstring: "{{ node_data.results[net.1].get_xml }}"
+    xpath: "//interface[@type='network']/source[@network='{{net.0.name}}']/../mac"
```

Comment:
so the problem is that sometimes type='bridge', maybe the following could work?
```
xpath: "//interface[@type='network' or @type='bridge']/source[@network='{{net.0.name}}']/../mac"
```

Reply:
I was thinking about that too, but as far as I can tell from how linchpin calls libvirt, it must contain a network definition (not bridge) and the reason we don't see it in the build is because there is a [leftover](https://jenkins-continuous-infra.apps.ci.centos.org/blue/organizations/jenkins/ci-linchpin/detail/ci-linchpin/1130/pipeline/#step-199-log-353) with wrong configurations but that task errors are ignored and we end up with misleading failure. I think we should add condition to not ignore errors